src/pages/Admin/Venue/Users/VenueUsersReport.tsx

```diff
@@ -0,0 +1,166 @@
+      if (userIsComplete(worldUsers[index])) {
+        setHeaders(Object.keys(worldUsers[index]));
+        return;
+      } else {
+        index++;
+      }
+    }
+  }, [worldUsers, userIsComplete]);
+
+  useEffect(() => {
+    getUsersHeaders();
+  }, [worldUsers, getUsersHeaders]);
+
+  const renderPageHeader = () => {
+    return <h4>Users Reports</h4>;
+  };
+
+  const renderUserTable = () => {
+    return (
+      <Table striped bordered hover>
+        <thead>
+          <tr>
+            <th>#</th>
+            <th>Party Name</th>
+            <th>Full Name</th>
+            <th>Department</th>
+          </tr>
+        </thead>
+        <tbody>
+          {/* when I try to use WithId<User> as realName are not added in it, it shows erro, so that's why I need any */}
+          {/* eslint-disable-next-line*/}
+          {worldUsers.map((user: any, index) => {
+            return (
+              userIsComplete(user) && (
+                <tr key={index}>
+                  <td>{index + 1}</td>
+                  <td>{user.partyName || "Not"}</td>
+                  <td>{user.realName || "Not set"}</td>
+                  <td>{user.companyDepartment || "Not set"}</td>
+                </tr>
+              )
+            );
+          })}
+        </tbody>
+      </Table>
+    );
+  };
+
+  const downloadCSV = useCallback(() => {
+    downloadGeneratedCSVFile(worldUsers, headersSelected);
+  }, [worldUsers, headersSelected]);
+
+  const renderDownloadBtn = () => {
+    return (
+      <button
+        className="btn btn-primary download-btn"
+        onClick={() => downloadCSV()}
+      >
+        Download{" "}
+      </button>
+    );
+  };
+
+  const updateSelectedHeaders = useCallback(
+    (selected: string) => {
+      if (headersSelected.includes(selected)) {
+        headersSelected.splice(headersSelected.indexOf(selected), 1);
+        setHeadersSelected([...headersSelected]);
+      } else {
+        setHeadersSelected([...headersSelected, selected]);
+      }
+    },
+    [headersSelected]
+  );
+
+  const addCustomField = useCallback(() => {
+    if (!headers.includes(customFieldValue)) {
+      setHeaders([...headers, customFieldValue]);
+    }
+  }, [customFieldValue, headers]);
+
+  const updateCustomFieldValue = useCallback(
+    (field: string) => {
+      setCustomFieldValue(field);
+    },
+    [setCustomFieldValue]
+  );
+
+  const renderFoundedHeaders = () => {
+    return (
+      <div className="venue-users-detail-filter-box-headers">
+        {headers.map((header) => (
+          <div
+            className="venue-users-detail-filter-box-headers-item"
+            key={header}
+          >
+            <Form.Check
+              type="switch"
+              id={header}
+              label={header}
+              checked={headersSelected.includes(header)}
+              onChange={() => updateSelectedHeaders(header)}
+            />
+          </div>
+        ))}
+      </div>
+    );
+  };
```

Comment:
Don't use render functions.

What's the intent of this code?

The `.map` should be `useMemo`'d

Reply:
fixed in 12e5c07fea64a78aced31d706faf59c5e523c733